libs/server/Resp/Bitmap/BitmapCommands.cs

```diff
@@ -450,19 +459,34 @@ private bool StringBitOperation<TGarnetApi>(int count, byte* ptr, BitmapOperatio
             if (NetworkKeyArraySlotVerify(ref keys, false))
                 return true;
 
-            if (sizeof(byte*) * (keyCount - 1) > 512)
+            // Match bitmap operation option
+            ConvertUtils.MakeUpperCase(optionSeq);
+            var bitop = optionSeq switch
             {
-                throw new Exception("Bitop source key limit (64) exceeded");
+                _ when optionSeq.SequenceEqual("OR"u8) => BitmapOperation.OR,
+                _ when optionSeq.SequenceEqual("AND"u8) => BitmapOperation.AND,
+                _ when optionSeq.SequenceEqual("XOR"u8) => BitmapOperation.XOR,
+                _ when optionSeq.SequenceEqual("NOT"u8) => BitmapOperation.NOT,
+                _ => BitmapOperation.NONE
+            };
+
+            // Check if option provided is valid
+            if (bitop == BitmapOperation.NONE)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_SYNTAX_ERROR, ref dcurr, dend))
+                    SendAndReset();
+                return true;
             }
 
-            var status = storageApi.StringBitOperation(keys, bitop, out long result);
-
-            if (status != GarnetStatus.NOTFOUND)
+            if (sizeof(byte*) * (keyCount - 1) > 512)
             {
-                while (!RespWriteUtils.WriteInteger(result, ref dcurr, dend))
-                    SendAndReset();
+                throw new Exception("Bitop source key limit (64) exceeded");
```

Comment:
What does this exception surface to client code as?

Reply:
Good point. I added a RESP response that is surfaced to the client instead of throwing an exception.